src/Access/ContextAccess.h

```diff
@@ -63,7 +63,7 @@ struct ContextAccessParams
 };
 
 
-class ContextAccess
+class ContextAccess : std::enable_shared_from_this<ContextAccess>
```

Comment:
See https://en.cppreference.com/w/cpp/memory/enable_shared_from_this:

```
public inheritance is mandatory
```

That's a bit tricky rule. But if the inheritance is not public then `weak_from_this()` always returns nullptr. And that's why the tests have failed.